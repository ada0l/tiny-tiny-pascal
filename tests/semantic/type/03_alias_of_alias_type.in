type int = Integer;
type int_int = int;
var a: int_int;
begin end.
