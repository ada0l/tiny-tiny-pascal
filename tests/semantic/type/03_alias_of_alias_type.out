main
   without name
   decls
      def types
         type def
            int
            type
               integer
      def types
         type def
            int_int
            type
               int
      vars
         var
            a
            type
               int_int
         without exp
   block
      compounds statements

------------------------------------------------------------------------------------------------------------------------------------------------------
| id                          | type of object              | ret type                                                                               |
------------------------------------------------------------------------------------------------------------------------------------------------------
| boolean                     | type                        | boolean                                                                                |
| integer                     | type                        | integer                                                                                |
| double                      | type                        | double                                                                                 |
| char                        | type                        | char                                                                                   |
| string                      | type                        | string                                                                                 |
| writeln                     | procedure                   | writeln                                                                                |
| write                       | procedure                   | write                                                                                  |
| read                        | procedure                   | read                                                                                   |
| random                      | function                    | double                                                                                 |
======================================================================================================================================================
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
| sqrt                        | function                    | double                                                                                 |
======================================================================================================================================================
| x                           | param                       | double                                                                                 |
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
------------------------------------------------------------------------------------------------------------------------------------------------------
------------------------------------------------------------------------------------------------------------------------------------------------------
| id                          | type of object              | ret type                                                                               |
------------------------------------------------------------------------------------------------------------------------------------------------------
| int                         | type alias                  | integer                                                                                |
| int_int                     | type alias                  | integer                                                                                |
| a                           | global var                  | integer                                                                                |
------------------------------------------------------------------------------------------------------------------------------------------------------
